Plugins~/Src/MeshSyncClientBlender/python/unity_mesh_sync_baking.py

```diff
@@ -0,0 +1,597 @@
+            matCopyIndex = bpy.data.materials.find(matCopyName)
+            if matCopyIndex >= 0:
+                matCopy = bpy.data.materials[matCopyIndex]
+            else:
+                mat.use_fake_user = True    # Make sure this does not get deleted when it's not referenced anymore
+                matCopy = mat.copy()
+                matCopy[ORIGINAL_MATERIAL] = mat.name
+                matCopy.name = f"{mat.name}_{obj.name}_baked"
+
+                print(f"Creating material copy '{mat.name}'->'{matCopy.name}'")
+
+                bakedBSDF = matCopy.node_tree.nodes.new(type='ShaderNodeBsdfPrincipled')
+
+                # Find lowest node in the tree and put the baked bsdf under that:
+                minYLocation = bsdf.location[1]
+                for node in matCopy.node_tree.nodes:
+                    minYLocation = min(minYLocation, self.getNodeYLocation(node))
+
+                bakedBSDF.location = (bsdf.location[0], minYLocation - 1000)
+                # Give bsdf a name and set its name on the material, so we can find it again:
+                bakedBSDF.name = BAKED_MATERIAL_SHADER
+                matCopy[BAKED_MATERIAL_SHADER] = bakedBSDF.name
+
+                matIndex = obj.material_slots.find(mat.name)
+                obj.material_slots[matIndex].material = matCopy
+            # obj.data.materials.pop(index=obj.data.materials.find(mat.name))
+            # obj.data.materials.append(matCopy)
+
+            mat = matCopy
+
+        self.ensureUVs(context, obj)
+
+        scene = context.scene
+
+        if obj.mode == 'EDIT':
+            bpy.ops.object.mode_set(mode='OBJECT')
+
+        scene.render.engine = "CYCLES"
+        scene.cycles.device = "GPU"
+        scene.cycles.samples = 10       # TODO: This could be a setting for the user to change
+        scene.render.bake.use_pass_direct = False
+        scene.render.bake.use_pass_indirect = False
+        scene.render.bake.use_pass_color = True
+        scene.render.image_settings.file_format = 'PNG'
+
+        # Set Correct Colour space for bake
+        scene.display_settings.display_device = 'sRGB'
+        scene.view_settings.view_transform = 'Standard'
+        scene.view_settings.look = 'None'
+        scene.view_settings.exposure = 0
+        scene.view_settings.gamma = 1
+        scene.sequencer_colorspace_settings.name = 'sRGB'
+        scene.cycles.time_limit = 0
+
+        # Set Cycles to use CUDA, if we aren't using any acceleration
+        if context.preferences.addons["cycles"].preferences.compute_device_type == "None":
+            context.preferences.addons["cycles"].preferences.compute_device_type = "CUDA"
+        scene.cycles.use_auto_tile = True
+        scene.cycles.max_bounces = 12
+        scene.cycles.diffuse_bounces = 8
+        scene.cycles.glossy_bounces = 2
+        scene.cycles.transparent_max_bounces = 0
+        scene.cycles.transmission_bounces = 2
+        scene.cycles.volume_bounces = 0
```

Comment:
Would it make sense to pass these values from the cycles panel?

Reply:
We're changing the settings on the cycles panel by setting those here.

Reply:
oh right, is this necessary though? Maybe its better to let the user configure these?

Reply:
also, if we decide to modify this through the script, should we revert the cycles settings after the bake?